test/fold_block.py

```diff
@@ -1,30 +1,72 @@
+import itertools
 import os
 
-next_block_id = 1
 
-
-class Fold(object):
+class BaseCiCfg:
+    _block_id_gen = itertools.count(1)
 
     def __init__(self):
-        global next_block_id
-        self.block_id = next_block_id
-        next_block_id += 1
-
-    def get_message(self, msg=''):
-        if os.environ.get('TRAVIS') == 'true':
-            if msg:
-                msg += ', '
-            msg += "see folded block '%s' above for details" % self.get_block_name()
+        self.block_id = next(self._block_id_gen)
+
+    @classmethod
+    def is_ci(cls) -> bool:
+        raise NotImplementedError("Use BaseCiCfg only as fallback")
+
+    def get_message(self, msg: str = "") -> str:
         return msg
 
-    def get_block_name(self):
-        return 'block%d' % self.block_id
+    def _get_message_folded(self, msg: str = "") -> str:
+        if msg:
+            msg += ", "
+        msg += "see folded block '%s' above for details" % self.get_block_name()
+        return msg
+
+    def get_block_name(self) -> str:
+        return "block%d" % self.block_id
 
     def __enter__(self):
-        if os.environ.get('TRAVIS') == 'true':
-            print('travis_fold:start:%s' % self.get_block_name())
         return self
 
     def __exit__(self, type, value, traceback):
-        if os.environ.get('TRAVIS') == 'true':
-            print('travis_fold:end:%s' % self.get_block_name())
+        pass
+
+
+class TravisCiCfg(BaseCiCfg):
+    @classmethod
+    def is_ci(cls) -> bool:
+        return os.environ.get("TRAVIS") == "true"
+
+    def get_message(self, msg=""):
+        return self._get_message_folded(msg)
+
+    def __enter__(self):
+        print("\ntravis_fold:start:%s" % self.get_block_name())
+        return self
+
+    def __exit__(self, type, value, traceback):
+        print("\ntravis_fold:end:%s" % self.get_block_name())
+
+
```

Comment:
This bit can be removed; we are not going to go back to `travis`.

Reply:
I first thought to delete it, but it doesn't hurt being there, so...
@audrow what do you think?

Reply:
If we're not using it, I'd favor removing it.

The reason is that I'd rather not have unused code hanging around.